File: algorithms/0594.longest_harmonious_subsequence/longest_harmonious_subsequence.rs
use std::collections::HashMap;
impl Solution {
    
    fn find_lhs(nums: Vec<i32>) -> i32 {
        let mut maximum = 0;

        let mut number_counter = HashMap::new();
        for n in nums.iter() {
            let mut counter = number_counter.entry(n).or_insert(0);
            *counter += 1;
        }

        for &n in nums.iter() {
            if number_counter.contains_key(&(n + 1)) {
                maximum = maximum.max(number_counter[&n] + number_counter[&(n + 1)]);
            }
        }
        maximum
    }
}

File: algorithms/0594.longest_harmonious_subsequence/longest_harmonious_subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int result = 0;
        for (int i = 0, j = 0; j < nums.size(); ++j) {
            while (i < j && nums[j] - nums[i] > 1) ++i;
            if (nums[j] - nums[i] == 1) {
                result = max(result, j - i + 1);
            }
        }
        return result;
    }
};